### PR TITLE
feat: invoke TQL via SQL interface

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Impl #986 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datanode/src/instance/sql.rs [hunk 1]
```diff
@@ -198,6 +199,33 @@ impl Instance {
...
+                let stmt = QueryLanguageParser::parse_promql(&promql).context(ExecuteSqlSnafu)?;
+                let logical_plan = self
+                    .query_engine
+                    .statement_to_plan(stmt, query_ctx)
+                    .context(ExecuteSqlSnafu)?;
+
+                self.query_engine
+                    .execute(&logical_plan)
+                    .await
+                    .context(ExecuteSqlSnafu)
```
Comment (MichaelScofield) [inline src/datanode/src/instance/sql.rs:224]: `ExecuteSqlSnafu` => `ExecuteTqlSnafu`?

File: src/sql/src/parsers/tql_parser.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    fn parse_string_or_number(
+        parser: &mut Parser,
+        token: Token,
+    ) -> std::result::Result<String, ParserError> {
+        let value = match parser.next_token() {
+            Token::Number(n, _) => n,
+            Token::DoubleQuotedString(s) | Token::SingleQuotedString(s) => s,
+            unexpected => {
+                return Err(ParserError::ParserError(format!(
+                    "Expect number or string, but is {unexpected}"
```
Comment (MichaelScofield) [inline src/sql/src/parsers/tql_parser.rs:81]: debug output to carry type:

```suggestion
                    "Expect number or string, but is {unexpected:?}"
```

**Review by MichaelScofield — APPROVED** [review]

File: src/query/src/datafusion/planner.rs [hunk 3]
```diff
@@ -93,6 +93,7 @@ where
         match statement {
             Statement::Query(qb) => self.query_to_plan(qb),
             Statement::Explain(explain) => self.explain_to_plan(explain),
+            Statement::Tql(_tql) => todo!(),
```
Comment (waynexia) [inline src/query/src/datafusion/planner.rs:96]: This should be unreachable.

I'm wondering do we really need to add a `Tql` variant in `Statement`? `Statement` is for SQL, though TQL is sit on SQL's interface. Giving `Tql` a dedicated enum or using promql's `Expr` would be clearer IMO.

File: src/sql/src/parsers/tql_parser.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+/// TQL extension parser, including:
+/// - TQL EVAL <query>
+/// - TQL EXPLAIN <query>
+impl<'a> ParserContext<'a> {
+    pub(crate) fn parse_tql(&mut self) -> Result<Statement> {
+        self.parser.next_token();
+
+        match self.parser.peek_token() {
+            Token::Word(w) => match w.keyword {
+                Keyword::NoKeyword if w.value.to_uppercase() == EVAL && w.quote_style.is_none() => {
```
Comment (waynexia) [inline src/sql/src/parsers/tql_parser.rs:39]: Do we need to support `EVALUATE` as well? Like `DESC` and `DESCRIBE`

File: src/datanode/src/instance/sql.rs (hunk 1 above)
Comment (killme2008) [reply src/datanode/src/instance/sql.rs:224]: The TQL is still a SQL language, just an extension.

File: src/sql/src/parsers/tql_parser.rs (hunk 4 above)
Comment (killme2008) [reply src/sql/src/parsers/tql_parser.rs:39]: Agree, I'll add it.

File: src/query/src/datafusion/planner.rs (hunk 3 above)
Comment (killme2008) [reply src/query/src/datafusion/planner.rs:96]: I think the `TQL` is just like `Query` variant ; it's another query statement.

**Comment by killme2008** [comment]
@waynexia  PTAL

**Review by waynexia — APPROVED** [review]
Looks fantastic to me 🚀

**Pull request merged**